### DISCUSSION TITLE
Customization of the build command

### DISCUSSION THREAD

**Discussion by mtribiere** [xmake-io/xmake#3737] [Q&A]
CMake offers this kind of command customization: 
```
set(CMAKE_C_COMPILE_OBJECT  "<CC> <FLAGS> -c -o <OBJECT>")
```
https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILE_OBJECT.html
This allows to describe exactly how we want the command to look like (Very useful for the embedded world).

Is there any equivalent in XMake?

**Comment by waruqi** [comment]
you can write on_buildcmd_file in custom rule to override builtin build command, then use os.exec to call gcc to build it.

https://github.com/xmake-io/xmake/blob/643719b7e52a0a2bf458853d2b0381e416bdf433/xmake/rules/lex_yacc/yacc/xmake.lua#L49

**Reply by mtribiere** [reply]
I'm not sure to understand, could you give me an example of this ?
